### PR TITLE
[JFG-902] - builders.

### PR DESCRIPTION
**Pull request by SokolAndrey**

### CODE REVIEW

**Comment by anton-antonenko** [comment]
:+1:

**Comment by a-badaev** [comment]
+1

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDescription.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+         * @param comment the comment of the test prototype.
+         */
+        public Builder withComment(String comment) {
+            this.comment = comment;
+            return this;
+        }
+
+        /**
+         * Sets end points for the tests inherited from the current test prototype.
+         *
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDescription.java]: Sets end points (where load will be applied during performance test) for the tests using this test prototype

Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDescription.java]: + see JHttpEndpoint for example

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDescription.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+         * @param endpointsProvider iterable end points. {@link java.util.List}, for example.
+         */
+        public Builder withEndpointsProvider(Iterable endpointsProvider) {
+            this.endpoints = endpointsProvider;
+            return this;
+        }
+
+
+        /**
+         * Sets queries for the tests inherited from the current test prototype.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDescription.java]: same as above

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestGroup.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    public static class Builder {
+        private String id;
+
+        private List<JTest> tests;
+
+        private Builder() {
+        }
+
+        /**
+         * Sets the tests in the group.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestGroup.java]: Sets the tests executed in parallel

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestSuite.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+
+    public static class Builder {
+        private List<JTestGroup> testGroups;
+
+        private Builder() {
+        }
+
+        /**
+         * Sets the {@code testGroups} for test suite.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestSuite.java]: + test groups will be executed in the order from beginning to the end of the list

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/JLoadRps.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
+package com.griddynamics.jagger.user.test.configurations.load;
+
+/**
+ * This type of load imitates an exact number of requests per second performed by Jagger.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/JLoadRps.java]: imitates => implements

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/JLoadRps.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+ * The number of requests, the number of threads and the warm up time values are configurable.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/JLoadRps.java]: number of threads => max allowed number of load threads

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/JLoadRps.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    private long warmUpTimeInSeconds;
+
+    private JLoadRps(Builder builder) {
+        this.maxLoadThreads = builder.maxLoadThreads;
+        this.requestsPerSecond = builder.requestsPerSecond;
+        this.warmUpTimeInSeconds = builder.warmUpTimeInSeconds;
+    }
+
+    /**
+     * This type of load imitates an exact number of requests per second. Where request is invoke from Jagger.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/JLoadRps.java]: imitates => generates

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/termination/JTerminationDuration.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
+package com.griddynamics.jagger.user.test.configurations.termination;
+
+/**
+ * Limits test execution time.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/termination/JTerminationDuration.java]: Test with such termination strategy will be executed for the defined time - duration

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/termination/JTerminationIterations.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
+package com.griddynamics.jagger.user.test.configurations.termination;
+
+/**
+ * Limits the number of execution iterations of a test.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/termination/JTerminationIterations.java]: Test with such termination strategy will last till defined number of requests are executed

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/termination/JTerminationIterations.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        return new Builder();
+    }
+
+
+    public static class Builder {
+        private long iterationCount;
+        private long maxDurationInSeconds;
+
+        /**
+         * Sets target number of test executions.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/termination/JTerminationIterations.java]: of requests to the system under test

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/termination/JTerminationIterations.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+         *
+         * @param iterationCount the number of test executions.
+         */
+        public Builder withIterationsCount(long iterationCount) {
+            this.iterationCount = iterationCount;
+            return this;
+        }
+
+        /**
+         * Sets the maximum test execution time in seconds.
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/termination/JTerminationIterations.java]: sets timeout for the test execution

**Pull request merged**